pkg/core/blockchain.go

```diff
@@ -454,6 +455,27 @@ func (bc *Blockchain) JumpToState(p uint32) error {
 		return fmt.Errorf("can't perform MPT jump to height %d: %w", p, err)
 	}
 
+	b := bc.dao.Store.Batch()
+	bc.dao.Store.Seek([]byte{byte(storage.STStorage)}, func(k, _ []byte) {
+		// Must copy here, #1468.
+		key := slice.Copy(k)
+		b.Delete(key)
+	})
+	bc.dao.Store.Seek([]byte{byte(storage.STTempStorage)}, func(k, v []byte) {
+		// Must copy here, #1468.
+		oldKey := slice.Copy(k)
+		b.Delete(oldKey)
+		key := make([]byte, len(k))
+		key[0] = byte(storage.STStorage)
+		copy(key[1:], k[1:])
+		value := slice.Copy(v)
+		b.Put(key, value)
+	})
+	err = bc.dao.Store.PutBatch(b)
```

Comment:
Whole state in a batch?

Reply:
I'm not sure about the right way to do it. Should I manually check the batch size and periodically flush it?